### ISSUE TITLE
Microsoft support

### ISSUE THREAD

**Issue by wyvest** [MultiMC/MultiMC5#3588]
System Information
-----------------------------
MultiMC version: 0.6.11-1430

Operating System: Windows 10 build 2004

Summary of the issue or suggestion: There is a pretty easy way to implement Microsoft support
----------------------------------------------

What should happen: When the code is triggered, a Microsoft login screen will show up and the user can login into mc
------------------------------

Steps to reproduce the issue (Add more if needed):
-------------------------------------------------------------
1. https://wiki.vg/Microsoft_Authentication_Scheme is how to do it

2.

3.

Suspected cause:
---------------------------
n/a

Logs/Screenshots:
----------------------------
[//]: # (Please refer to https://github.com/MultiMC/MultiMC5/wiki/Log-Upload for instructions on how to attach your logs.)

Additional Info:
---------------------------
pretty easy to do please fix

**Comment by phit** [comment]
## Summary of the the reply: There is a pretty easy way to find out why this hasn't happened yet

## What should happen: When you make an issue, use the Github search before you do

## Steps to reproduce the issue

1. https://github.com/MultiMC/MultiMC5/issues?q=is%3Aissue+microsoft+

## Suspected cause:

you didn't bother searching

## Logs/Screenshots:
## Additional Info:

pretty easy to do please fix

**Comment by Zetabite** [comment]
No

**Comment by wyvest** [comment]
> No

:(

**Comment by seishin4real** [comment]
@Zetabite, would you consider merging a pull req. if @wyvest provided one?
@wyvest, would you consider submitting a PR if @Zetabite says they'd give it a try?

**Comment by Zetabite** [comment]
no

**Comment by Zetabite** [comment]
besides that i am not even developing for mmc. If one implements a solution, its peterix and non of this bs.

**Comment by Zetabite** [comment]
and you would know that this wiki entry was denied to be used by peterix several times in closed issues and discord convos

**Comment by seishin4real** [comment]
lol what? how would I know any of this (or care for that matter)? my son has just bought java edition and can't use multimc, so I googled a bit and found this.
wyvest says it's easy, so if he could provide the implementation why not use it?!..

btw. if you're not part of the dev team why are you making statements?

**Comment by Zetabite** [comment]
cause i am in the discord and we get this convo constantly. The solution will be provided by peterix.

**Comment by Zetabite** [comment]
And if you would have read through any of the topic related issues, you would have known it was denied before.

**Comment by Zetabite** [comment]
I excuse, barely any issues has this link and talk about, i messed up there. But still, on the discord we often get the same wiki entry and it was denied often.

**Comment by Zetabite** [comment]
So unless peterix plans to change their mind about using is, this wont be an accepted solution.

**Comment by Zetabite** [comment]
ok, sometimes i should just shut up:
`- TODO: MSA implementation
TBD, this will most likely be implemented as a standalone program + library, then integrated once it's stable
Waiting on internal Mojang decisions and public statement
Likely approach is detailed at https://wiki.vg/Microsoft_Authentication_Scheme
Considering to use https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#device-code
Account migration needs to be handled properly
`

**Comment by kb-1000** [comment]
IIRC it was said to use ATLauncher's approach to it. May or may not be the same thing. Either way, the implementation will be from peterix.

**Comment by Zetabite** [comment]
> my son has just bought java edition and can't use multimc

what your son belongs, hope they can enjoy multimc soon.

**Comment by jikuja** [comment]
> There is a pretty easy way to implement Microsoft support

I would not call it easy if it requires implementing full web browser with javascript support. 

Support for MSA is on the TODO list and will be done at some point. Meanwhile, there is no need to link well-known resources and explain how easy it would be to implement by copying the logic of this and that.

**Comment by Janrupf** [comment]
I want to sum this up before it gets even more heated.

Yes, a lot of people have looked into implementing MSA for MultiMC, and sadly it is not as easy. Microsoft has yet to provide an at least somewhat official way, before that MultiMC probably wont get MSA (unless it takes too long and they start migration of accounts before an official way). In the end Peterix decides what happens, and he has provided a lot of reasons why it currently does not make sense to implement MSA. It will happen as soon as there is a way which might not break or violate anything.

I'm pretty sure someone will create a PR (or Peterix will implement it himself) when above conditions are matched. Please do not waste your time on doing this currently.

**Comment by peterix** [comment]
This does not provide any new information or any value.

Thank you for your concern, but I've been involved with implementing MSA in the vanilla launcher, and I will add it to MMC once it's ready.
